simmechanics/CMakeLists.txt

```diff
@@ -93,28 +93,96 @@ macro(generate_icub_simmechanics)
     else()
       set(GAZEBO_ASSIGNED_INERTIAS "")
     endif()
+
+    # If we are generating the icub2.5+ model we need to add the xsens IMU, change the mesh folder
+    # and change the rotation axis list that need to be reversed.
+    if(GIVTWO_ICUB_PLUS)
+      set(MESH_FOLDER "filenameformatchangeext: \"package://iCub/meshes/simmechanics/2-5_plus/%s-binary.stl\"")
+      set(ANKLE_PITCH_ROM "-50,20")
+      set(CUSTOM_EPSILON "epsilon: 2e-7")
+      set(XSENS_IMU_FRAME
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    frameReferenceLink: root_link
+    exportedFrameName: root_link_imu_frame
+")
+      set(XSENS_IMU_SENSOR
+"  - frameName: SCSYS_ROOT_LINK_XSENS_IMU
+    linkName: root_link
+    exportFrameInURDF: No
+    sensorName: root_link_xsens_imu
+    sensorType: \"imu\"
```

Comment:
Actually, while the additional frame (`@XSENS_IMU_FRAME@`) exported for the IMU is required for the dynamic computations in iDynTree, the IMU itself is not considered as a URDF-like "sensor" as per the description in https://github.com/robotology/idyntree/blob/master/doc/model_loading.md and doesn't require an entry in the "sensor" tag section. The following entry...
```
  - frameName: SCSYS_HEAD_MTX_IMU
    linkName: head
    sensorName: head_imu_acc_1x1
    sensorType: "accelerometer"
    ...
```
was already there and was intended to provide information on the accelerometers embedded in the head IMU. those information were used for the sensor measurement prediction computation in iDynTree.

This said, we should have for the root_link IMU:
```
    sensorName: root_link_imu_acc
    sensorType: "accelerometer"
```

Reply:
I agree with nuno on this.

Reply:
Updated!